Tools/gsc/test/makefile

```diff
@@ -0,0 +1,138 @@
+TESTCASES?=python3 hello-world nodejs bash numpy pytorch
+DISTRIBUTIONS?=ubuntu18.04 ubuntu16.04
+TESTS=$(foreach D,$(DISTRIBUTIONS),$(foreach T,$(TESTCASES),$D-$T))
+SGX_DEVICES=--device=/dev/gsgx --device=/dev/isgx -v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket
+
+.PHONY: test clean cleanBase cleanGSC cleanContainer
+
+IMAGE_SUFFIX?=
+
+all: $(TESTS)
+	for d in $(DISTRIBUTIONS); do \
+		sed 's/\"distro\": \".*\"/\"distro\": \"'$${d}'\"/' ../config.json.template > ../config.json; \
+		$(MAKE) $(addprefix gsc-$${d}-, $(TESTCASES)) || exit 1; \
+	done
+
+%: %.dockerfile
+	echo "Build base image $@.dockerfile"
+	docker build -t $(addsuffix $(IMAGE_SUFFIX), $@) -f $@.dockerfile .
+
+gsc-%-bash:
+	echo "Build graphenized image $@"
+	cd .. && ./gsc build $(addsuffix $(IMAGE_SUFFIX), $*-bash) bash.manifest ls.manifest
+
+gsc-%-python3: %-python3.manifest
+	echo "Build graphenized image $@"
+	cd .. && ./gsc build $(addsuffix $(IMAGE_SUFFIX), $*-python3) $(addprefix test/, $^ sh.manifest ls.manifest)
+
+gsc-%-pytorch: %-pytorch.manifest
+	echo "Build graphenized image $@"
+	cd .. && ./gsc build $(addsuffix $(IMAGE_SUFFIX), $*-pytorch) $(addprefix test/, $^ sh.manifest uname.manifest)
+
+gsc-%:
+	echo "Build graphenized image $@"
+	cd .. && ./gsc build $(addsuffix $(IMAGE_SUFFIX), $*) test/$(*:gsc-%=%).manifest
+
+test: $(addprefix test-, $(DISTRIBUTIONS))
+	echo "[SUCCESS] Completed all GSC test cases"
+
+MAXTESTNUM?=11
+
+test-1-%:
+	docker run $(SGX_DEVICES) $(addsuffix $(IMAGE_SUFFIX), gsc-$*-python3) -c 'print("HelloWorld!")' >out 2>/dev/null;
+	grep -q "HelloWorld!" out;
+	rm -f out
+
+test-2-%:
+	docker run $(SGX_DEVICES) $(addsuffix $(IMAGE_SUFFIX), gsc-$*-python3) /graphene-tests/python-simple/scripts/helloworld.py >out 2>/dev/null
+	grep -q "Hello World" out;
+	rm -f out
+
+test-3-%:
+	docker run $(SGX_DEVICES) $(addsuffix $(IMAGE_SUFFIX), gsc-$*-python3) /graphene-tests/python-simple/scripts/fibonacci.py >out 2>/dev/null;
+	grep -q "fib2              55" out;
+	rm -f out
+
+test-4-%:
+	docker run $(SGX_DEVICES) $(addsuffix $(IMAGE_SUFFIX), gsc-$*-python3) -c 'import os;os.system("ls")' >out 2>/dev/null;
```

Comment:
When creating gsc-ubuntu18.04-python3 image, in the manifest file, you have sh and ls trusted children, and I suppose it is for this test case-> import os;os.system("ls"). But, I was able to run os.system("pwd"), without having pwd as trusted-child in the manifest.

Reply:
Also this may be generic. In the sense, you can remove ls as a trusted child, and check if os.system("ls") works fine.

Reply:
This is due to the shell serving the command itself. Please refer to the the shell [command search and execution](https://pubs.opengroup.org/onlinepubs/9699919799//utilities/V3_chap02.html#tag_18_09_01_01) details (1.d. shows pwd is part of built-in tools). 

Since pwd is built-in, sh will print the working directory to stdout without creating another process. This is different than the behavior for ls which is not built-in and causes a new process to be created which requires a manifest file.